src/event/service/event.service.ts

```diff
@@ -0,0 +1,108 @@
+import mongoose from "mongoose";
+import { NotFoundError } from "../error/response/not-found.error";
+import { getFetchHeaders, microserviceUrl } from "../helper/microservice.url";
+import { Event } from "../schema/db/event.schema";
+import type { TFilterEventsValidator, TUserEventsValidator } from "../schema/request/event.schema";
+
+export async function getFilteredEvents(data: TFilterEventsValidator, userId: string) {
+    const location = data.location;
+
+    const dateStart = data.dateStart;
+    const dateEnd = data.dateEnd;
+
+    const requiredRating = Number(data.rating);
+
+    const type = data.type;
+
+    const friendsOnly = data.friendsOnly;
+
+    const publicOnly = data.publicOnly;
+
+    const pageSize = Number(data.pageSize);
+    const pageNumber = Number(data.pageNumber);
```

Comment:
Nemusíš tu mít ty number() wrappingy, protože ti to Zod přes to coerce zparsuje na number samo :) 

Za další nemusíš takhle zdlouhavě vypisovat variables - můžeš udělat `const { location, dateStart, dateEnd, ... } = data;`, čímž si rovnou najednou vytvoříš ty proměnné z toho objektu. :)